src/containers/paint-editor.jsx

```diff
@@ -38,14 +40,37 @@ class PaintEditor extends React.Component {
             'canRedo',
             'canUndo',
             'handleCopyToClipboard',
-            'handlePasteFromClipboard'
+            'handlePasteFromClipboard',
+            'setPaintEditor',
+            'onMouseDown',
+            'startEyeDroppingLoop',
+            'stopEyeDroppingLoop'
         ]);
+        this.state = {
+            colorInfo: null
+        };
     }
     componentDidMount () {
         document.addEventListener('keydown', this.props.onKeyPress);
     }
+    shouldComponentUpdate (nextProps, nextState) {
```

Comment:
Did you need to add this to optimize for something in particular?

Reply:
for optimizing the `setInterval` method to not re-render every time the color info is changed

Reply:
It's checking this.state.colorInfo !== nextState.colorInfo so wouldn't it still rerender?

Reply:
This handles the case of a) hovering outside of the confines of the canvas, or b) if the loupe is paused in a particular place for a while. If we're not too concerned with performance in this case, then I can take it out though

Reply:
Can we check the colorInfo equality before we call setState instead of skipping the rendering in shouldComponentUpdate?